coordinator/internal/logic/provertask/prover_task.go

```diff
@@ -121,6 +121,11 @@ func (b *BaseProverTask) hardForkSanityCheck(ctx *gin.Context, taskCtx *proverTa
 		return "", getHardForkErr
 	}
 
+	// for backward compatibility, darwin chunk prover can still prove darwinV2 chunk tasks
+	if taskCtx.taskType == message.ProofTypeChunk && hardForkName == "darwin" {
```

Comment:
I think here is wrong. current chunk must be darwinV2

Reply:
fixed in [2e7827e](https://github.com/scroll-tech/scroll/pull/1640/commits/2e7827e3554bde6525044b0f0360ac8ca84bd613).